File: src/types/messages.ts
export type Conversation = {
  id: string;
  other_user: {
    id: string;
    full_name: string;
    profile_picture_url: string | null;
  };
  last_message: string | null;
  last_message_at: string | null;
  unread_count: number;
};

export type DirectMessage = {
  id: string;
  conversation_id: string;
  sender_id: string;
  content: string;
  created_at: string;
  read_at: string | null;
  is_mine: boolean;
};

export type CanMessage = {
  can_message: boolean;
  reason: string | null;
};

File: backend/app/schemas/messages.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ConversationCreate(BaseModel):
    other_user_id: UUID


class MessageCreate(BaseModel):
    content: str = Field(min_length=1, max_length=2000)


class MessageOut(BaseModel):
    id: UUID
    conversation_id: UUID
    sender_id: UUID
    content: str
    created_at: datetime
    read_at: datetime | None
    is_mine: bool

    model_config = {"from_attributes": True}


class ConversationParticipant(BaseModel):
    id: UUID
    full_name: str
    profile_picture_url: str | None

    model_config = {"from_attributes": True}


class ConversationOut(BaseModel):
    id: UUID
    other_user: ConversationParticipant
    last_message: str | None
    last_message_at: datetime | None
    unread_count: int


class CanMessageOut(BaseModel):
    can_message: bool
    reason: str | None = None
